docs/src/main/asciidoc/security-openid-connect-multitenancy.adoc

```diff
@@ -844,6 +844,38 @@ The default tenant configuration is automatically disabled when `quarkus.oidc.au
 
 Note that tenant specific configurations can also be disabled, for example: `quarkus.oidc.tenant-a.tenant-enabled=false`.
 
+== Authentication outside of the HTTP request context
+
+Sometimes it is practical to be able to create `SecurityIdentity` for given token when there is no active request context.
+The `quarkus-oidc` extension uses its own `OidcIdentityProvider` to convert a token to a `SecurityIdentity` instance.
+When the provider is used during an HTTP request, the tenant configuration can be resolved as described in the <<static-tenant-resolution>> section.
+However, when there is no active HTTP request, you need to select tenant explicitly with the `io.quarkus.oidc.TenantFeature` qualifier like in the example below:
+
+[source,java]
+----
+package org.acme.quickstart.oidc;
+
+import io.quarkus.oidc.TenantFeature;
+import io.quarkus.oidc.runtime.OidcIdentityProvider;
+import io.quarkus.runtime.StartupEvent;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.smallrye.mutiny.Uni;
+import jakarta.enterprise.event.Observes;
+
+public class StartupService {
+
+    void init(@Observes StartupEvent event, @TenantFeature("tenantId") OidcIdentityProvider identityProvider) {
```

Comment:
It looks fine, but I'd like to propose to have an example showing what seems to be a more typical case, the request comes in, the token is passed to some other thread, the request has gone, and the new runner verifies the token

Reply:
Please keep this example, but show the proposed example first

Reply:
Charm of this example is that everyone will understand it straight away, while some threading stuff is going to be complicated, they will need to think and hope they figure what I meant. 

**Please don't get me wrong, I'll do it if you are convinced of this**, but before let me to provide few points to which you can react:

- it is very close to scheduled tasks https://quarkus.io/guides/scheduler I suppose user will use service account where possible, but #24069 explicitly mentioned scheduled methods
- > more typical case, the request comes in, the token is passed to some other thread, the request has gone, and the new runner verifies the token

if the only problem is that HTTP request passed, can you leverage context propagation instead of this? I take it there, `RoutingContext` will still be stored in the CDI request context, so it seems to me that I would be suggesting something that users shouldn't do.

- I think the point here is to be able to use `OidcIdentityProvider` independently on HTTP request, do people really store old access token in real world situation from ended HTTP request and use it without the original CDI request context? I don't know, maybe there were users that already mentioned this?

Reply:
There is one else thing @sberyozkin  and please push back, I'm just asking: why would users really store old tokens instead of injecting identity when request is active and use it in the new thread when necessary?

Reply:
sorry @sberyozkin I didn't mention following comment `Please keep this example, but show the proposed example first`. I'll do it..

Reply:
@michalvavrik Np, @knutwannheden may provide some context, and I also saw something similar asked in Discussions, #27754, etc

Reply:
Thinking more about it, `getAdminToken` - how would anyone implement it, using `OidcTestClient` ? The use case is to verify the access token which was left by the completed request, tokens acquired by clients in scope of the current endpoint would not be targeting this endpoint, they are meant to be propagated to downstream services (Keycloak, or some microservices). It is an interesting example, but IMHO it may confuse users as to the main goal of this enhancement

Reply:
>messing with threads is a bad idea

Why ? Users may have different use cases, though indeed, as I just commented above, if the token is passed out of band, then it is just a case of the manual verification, no threads are involved.

Reply:
Use case is legit, but documentation should be simple if possible. I'm not sure what is level of users knowledge on this topic and if it is possible to show example without using different executor, I'd do it.

Reply:
@michalvavrik So may be lets just show a manual verification of the token passed for example with the custom header, or as a JSON property ?

Reply:
done. I'll close this thread as I addressed suggestion and we many thread comments opened. Please re-open if you are not satisfied or create a new one.